starter/app/src/main/java/com/udacity/project4/locationreminders/geofence/GeofenceBroadcastReceiver.kt

```diff
@@ -3,21 +3,60 @@ package com.udacity.project4.locationreminders.geofence
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import android.util.Log
+import android.widget.Toast
+import com.udacity.project4.MyApp
+import com.udacity.project4.R
+import com.udacity.project4.locationreminders.reminderslist.ReminderDataItem
+import com.udacity.project4.locationreminders.savereminder.SaveReminderFragment.Companion.ACTION_GEOFENCE_EVENT
+import com.udacity.project4.locationreminders.savereminder.SaveReminderFragment.Companion.GEO_FENCE_ID
+import com.udacity.project4.locationreminders.savereminder.SaveReminderFragment.Companion.GEO_FENCE_LAT
+import com.udacity.project4.locationreminders.savereminder.SaveReminderFragment.Companion.GEO_FENCE_LNG
+import com.udacity.project4.utils.TAG
+import com.udacity.project4.utils.sendNotification
 
 /**
- * Triggered by the Geofence.  Since we can have many Geofences at once, we pull the request
- * ID from the first Geofence, and locate it within the cached data in our Room DB
+ * Triggered by the Geofence. Since we can have many GeoFences at once, we pull the request
+ * ID from the first Geofence, and locate it within the cached data in our Room DB.
  *
- * Or users can add the reminders and then close the app, So our app has to run in the background
+ * Or users can add the reminders and then close the app, so our app has to run in the background
  * and handle the geofencing in the background.
- * To do that you can use https://developer.android.com/reference/android/support/v4/app/JobIntentService to do that.
- *
+ * To do that you can use:
+ * https://developer.android.com/reference/android/support/v4/app/JobIntentService to do that.
  */
-
 class GeofenceBroadcastReceiver : BroadcastReceiver() {
-    override fun onReceive(context: Context, intent: Intent) {
 
-//TODO: implement the onReceive method to receive the geofencing events at the background
+	override fun onReceive(context: Context, intent: Intent) {
```

Comment:
@graffiti75 Kindly, where I can find the TODO: implement onRecieve method... in your updated code?

Reply:
New PR created: https://github.com/udacity/nd940-android-kotlin-c4-starter/pull/14